src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -167,32 +193,31 @@ private async Task PublishResourcesWithInitialStateAsync()
         }
     }
 
-    private void WatchResourceChanges(CancellationToken cancellationToken)
+    private void WatchResourceChanges()
     {
-        var semaphore = new SemaphoreSlim(1);
+        var outputSemaphore = new SemaphoreSlim(1);
 
-        Task.Run(
+        _resourceWatchTask = Task.Run(
```

Comment:
This `Task.Run` started in this method is awaited but the others aren't. Consider making them consistently awaited, or add a comment explaining why this one is different.